### DISCUSSION TITLE
Queuing command without waiting for the api response

### DISCUSSION THREAD

**Discussion by RoemIko** [CrowdStrike/falconpy#724] [Q&A] [answered]
Is it possible to just send commands to the CrowdStrike API and collect the response later? For example I send a command that runs an executable. But this executable needs time to run. I don't want to wait for this command so i wait for the timeout. I send a command after it like "collect the output of this executable AFTER its done".  So i send a runscript command followed by a get command through batch_admin_command

Thanks in advance,

**Comment by jshcodes** [comment, marked as answer]
Hi @Roemiko -

Yes there is!  

Due to the nature of the sample I'm working on for your [multi-tenant question](https://github.com/CrowdStrike/falconpy/discussions/717), I'll be speaking to offline queueing as part of the example for your response.  In the interim, check out [this sample](https://github.com/CrowdStrike/falconpy/blob/main/samples/rtr/queued_execute.py) to get an idea for how it works.  (The version I'm currently testing will demonstrate [body payload abstraction](https://www.falconpy.io/Usage/Payload-Handling.html#body-payload-abstraction) for the methods used as well.)

**Reply by jshcodes** [reply]
Side note: Since you're not wanting to wait for host execution, and the host is online, you could also stop processing and check the result using the `cloud_request_id` after a period of time.

**Reply by RoemIko** [reply]
if i do that does it wait for the previous command to finish? For example: running autorunsc, wait till its done then continue with the next command. Or does it just run them back to back without waiting for the previous commands to be done?

**Reply by jshcodes** [reply]
Commands are executed in the order they are received. Your long running command should complete before the second command is executed. 

Each command would have it's own `cloud_request_id` so you will be able to check them separately.

**Reply by RoemIko** [reply]
and how would i stop the processing the response just wait for the timeout or break after i shot in a command?

**Reply by RoemIko** [reply]
it appears that commands are not performed after they are done processing when a host is online. 

The problem im facing is that i dont want to wait for the output of a command, i just want to queue as many of these commands as i can and come back later to check the status of it. I know you can use `ListSessions` and `check_admin_command_status` with a `trace-id/cloud_request_id` to check the status later on. But im facing the issue that i have to wait interactively, is it possible to not wait interactively and queue the commands on an online host? or do i have to check interactively if a host is done with the said command?

```
script_file = open('commands.txt', 'r')
commands = script_file.readlines()
for arg in commands:
    response = falcon_rtra.batch_admin_command(base_command="runscript",
                            batch_id=sessions['body']['batch_id'],
                            command_string=arg,
                            persist_all=True,
                            timeout=30
                            )
```
in the commands.txt i run the following commands:
```
cd Windows/Temp
mkdir CSDeploy
cd CSDeploy
put autorunsc.exe
runscript -Raw=```cmd.exe /c autorunsc.exe -accepteula -nobanner -a *-vt > output.csv```
runscript -Raw=```Get-ChildItem -Recurse -Path C:\Windows\Temp\CSDeploy\output.csv | Copy-Item -Recurse -Destination C:\Windows\Temp\AutoFalcon\$env:computername.csv; Compress-Archive -Path C:\Windows\Temp\CSDeploy\$env:computername.csv  -DestinationPath C:\autorunsc.zip -Force;```
get C:\autorunsc.zip
rm C:\autorunsc.zip
cd c:/
rm -force Windows/Temp/CSDeploy
```

In the auditing i can see that these commands are run after eachother, but they dont seem to wait for eachother.

**Reply by jshcodes** [reply]
Hi @RoemIko -

Have you tried moving over to executing this via a script that you trigger from RTR?  You could upload the script as a CloudFile, then execute it from RTR.  I believe that might execute synchronously like you're wanting.

**Reply by RoemIko** [reply]
We have tried that, I contacted CrowdStrike support and they told is that its not possible to queue commands sadly.
